File: AgentieTurismJava/src/main/java/org/example/agentietursimjava/Domain/Rezervare.java
package org.example.agentietursimjava.Domain;

import java.util.Objects;
import java.util.UUID;

public class Rezervare extends Entity<UUID> {

    private String numeClient;
    private String numarTelefonClient;
    private int nrBilete;
    private Excursie excursie;

    public Rezervare(String numeClient, String numarTelefonClient, int nrBilete, Excursie excursie) {
        this.setId(UUID.randomUUID());
        this.numeClient = numeClient;
        this.numarTelefonClient = numarTelefonClient;
        this.nrBilete = nrBilete;
        this.excursie = excursie;
    }

    public String getNumeClient() {
        return numeClient;
    }

    public void setNumeClient(String numeClient) {
        this.numeClient = numeClient;
    }

    public String getNumarTelefonClient() {
        return numarTelefonClient;
    }

    public void setNumarTelefonClient(String numarTelefonClient) {
        this.numarTelefonClient = numarTelefonClient;
    }

    public int getNrBilete() {
        return nrBilete;
    }

    public void setNrBilete(int nrBilete) {
        this.nrBilete = nrBilete;
    }

    public Excursie getExcursie() {
        return excursie;
    }

    public void setExcursie(Excursie excursie) {
        this.excursie = excursie;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Rezervare rezervare)) return false;
        return nrBilete == rezervare.nrBilete && Objects.equals(numeClient, rezervare.numeClient) && Objects.equals(numarTelefonClient, rezervare.numarTelefonClient) && Objects.equals(excursie, rezervare.excursie);
    }

    @Override
    public int hashCode() {
        return Objects.hash(numeClient, numarTelefonClient, nrBilete, excursie);
    }

    @Override
    public String toString() {
        return "Rezervare{" +
                "numeClient='" + numeClient + '\'' +
                ", numarTelefonClient='" + numarTelefonClient + '\'' +
                ", nrBilete=" + nrBilete +
                ", excursie=" + excursie +
                '}';
    }
}

File: AgentieTursimCS/Domain/Rezervare.cs
#nullable enable
using System;

namespace AgentieTurismCS.Domain
{
    public class Rezervare : Entity<Guid>
    {
        private string _numeClient;
        private string _numarTelefonClient;
        private int _nrBilete;
        private Excursie _excursie;

        public Rezervare(string numeClient, string numarTelefonClient, int nrBilete, Excursie excursie)
        {
            Id = Guid.NewGuid();
            _numeClient = numeClient;
            _numarTelefonClient = numarTelefonClient;
            _nrBilete = nrBilete;
            _excursie = excursie;
        }

        public string NumeClient
        {
            get => _numeClient;
            set => _numeClient = value;
        }

        public string NumarTelefonClient
        {
            get => _numarTelefonClient;
            set => _numarTelefonClient = value;
        }

        public int NrBilete
        {
            get => _nrBilete;
            set => _nrBilete = value;
        }

        public Excursie Excursie
        {
            get => _excursie;
            set => _excursie = value;
        }

        public override bool Equals(object? obj)
        {
            if (this == obj) return true;
            if (!(obj is Rezervare rezervare)) return false;
            return _nrBilete == rezervare._nrBilete && _numeClient == rezervare._numeClient && _numarTelefonClient == rezervare._numarTelefonClient && _excursie.Equals(rezervare._excursie);
        }

        public override int GetHashCode()
        {
            return HashCode.Combine(_numeClient, _numarTelefonClient, _nrBilete, _excursie);
        }

        public override string ToString()
        {
            return $"Rezervare{{numeClient='{_numeClient}', numarTelefonClient='{_numarTelefonClient}', nrBilete={_nrBilete}, excursie={_excursie}}}";
        }
    }
}